internal/server/access_keys.go

```diff
@@ -58,3 +60,120 @@ func (a *API) CreateAccessKey(c *gin.Context, r *api.CreateAccessKeyRequest) (*a
 		AccessKey:         raw,
 	}, nil
 }
+
+// See docs/dev/api-versioned-handlers.md for a guide to adding new version handlers.
+func (a *API) addPreviousVersionHandlersAccessKey() {
+	type listAccessKeysRequest_v0_16_1 struct {
+		UserID      uid.ID `form:"user_id"`
+		Name        string `form:"name"`
+		ShowExpired bool   `form:"show_expired"`
+		api.PaginationRequest
+	}
+	type accessKeyV0_18_0 struct {
+		ID                uid.ID   `json:"id"`
+		Created           api.Time `json:"created"`
+		LastUsed          api.Time `json:"lastUsed"`
+		Name              string   `json:"name"`
+		IssuedForName     string   `json:"issuedForName"`
+		IssuedFor         uid.ID   `json:"issuedFor"`
+		ProviderID        uid.ID   `json:"providerID"`
+		Expires           api.Time `json:"expires"`
+		ExtensionDeadline api.Time `json:"extensionDeadline"`
+	}
+	newAccessKeyV0_18_0FromLatest := func(latest api.AccessKey) accessKeyV0_18_0 {
+		return accessKeyV0_18_0{
+			ID:                latest.ID,
+			Created:           latest.Created,
+			LastUsed:          latest.LastUsed,
+			Name:              latest.Name,
+			IssuedForName:     latest.IssuedForName,
+			IssuedFor:         latest.IssuedFor,
+			ProviderID:        latest.ProviderID,
+			Expires:           latest.Expires,
+			ExtensionDeadline: latest.InactivityTimeout,
+		}
+	}
+	newAccessKeyListResponseV0_18_0FromLatest := func(latest *api.ListResponse[api.AccessKey]) *api.ListResponse[accessKeyV0_18_0] {
+		if latest == nil {
+			return nil
+		}
+		oldResp := &api.ListResponse[accessKeyV0_18_0]{
+			PaginationResponse: latest.PaginationResponse,
+			Count:              latest.Count,
+			LastUpdateIndex:    latest.LastUpdateIndex,
+			Items:              make([]accessKeyV0_18_0, len(latest.Items)),
+		}
+		for i, item := range latest.Items {
+			oldResp.Items[i] = newAccessKeyV0_18_0FromLatest(item)
+		}
+		return oldResp
```

Comment:
This can use `api.NewListResponse`

```suggestion
		return api.NewListResponse(latest.Items, latest.PaginationResponse, 	func(newItem api.AccessKey, oldItem accessKeyV0_18_0) {
    			return newAccessKeyV0_18_0FromLatest(newItem)
		})
```

Reply:
It could, but using `NewListResponse` is actually what almost caused me to introduce a bug in the list grants endpoint. That handler doesn't set `LastUpdateIndex`.

Maybe we should have a function to copy `ListResponse` and change the type of the items.

Reply:
I created `CopyListResponse` which calls `NewListResponse` and sets the extra fields. I think this works well for now.

I have been finding `api.ListResponse` a bit difficult to work with. It forced us to put other metadata onto this struct, instead of being able to embed it where necessary.

I think we might want to consider having separate api structs for each list response, and have each of them embed a `ListMetadata` struct that provides the `Count` and `PaginationResponse` fields. That would let us embed other structs without overloading `ListResponse`. I created #3993 to track that change.